src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -69,6 +77,11 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
   public void addMessage(Message message) {
     messages.add(message);
     persistentStorageAgent.writeThrough(message);
+    Activity activity1 = new Activity(message);
+    activity1.setIsPublic(true);
+    if (activityStore != null) {
+      activityStore.addActivity(activity1);
+    }
```

Comment:
- Whenever activityStore == null, it means that we haven't initialized the an activitystore so we cannot use it or it will prompt errors like nullPointers. 
- Yes, some activities are public and some are private. That's why I have that function.

Reply:
Oh, okay. That should be very useful.

Also, so what what what be I guess an example of a public activity vs private? Would it be like admin activities vs regular user activities?

Reply:
In what cases will the `activityStore` be null?  It seems like we would always expect there to be an `activityStore`.

Reply:
I was having nullPointers with the UserStore. I added this security check into all the instances where I used activityStore. I will remove it from both messageStore and conversationStore.

Reply:
Hmm, were you able to figure out why they were happening in the UserStore?  Is there somewhere that you should be initializing the AcitivityStore but aren't?

Reply:
@dhgarrette By using getTestInstance() for the tests, the (message, user, conv) stores are not initializing the activity store. So I fixed the bug and got rid of the if-statement.